package.json

```diff
@@ -46,32 +46,32 @@
   },
   "homepage": "https://github.com/ethereum/solc-js#readme",
   "dependencies": {
-    "command-exists": "^1.2.8",
-    "commander": "^8.1.0",
-    "follow-redirects": "^1.12.1",
-    "js-sha3": "0.8.0",
-    "memorystream": "^0.3.1",
-    "semver": "^5.5.0",
-    "tmp": "0.0.33"
+    "command-exists": ">=1.1.0",
+    "commander": ">=8.0.0 <12.0.0",
+    "follow-redirects": ">=1.0.0",
+    "js-sha3": ">=0.8.0",
+    "memorystream": ">=0.3.0",
+    "semver": ">=5.0.0",
+    "tmp": ">=0.0.26"
   },
   "devDependencies": {
-    "@types/node": "^16.11.7",
-    "@types/semver": "^7.3.9",
-    "@types/tape": "^4.13.2",
-    "@types/tmp": "^0.2.3",
-    "@typescript-eslint/eslint-plugin": "^5.8.0",
-    "@typescript-eslint/parser": "^5.8.0",
-    "coveralls": "^3.0.0",
-    "eslint": "^7.32.0",
-    "eslint-config-standard": "^16.0.3",
-    "eslint-plugin-import": "^2.25.3",
-    "eslint-plugin-node": "^11.1.0",
-    "eslint-plugin-promise": "^5.1.1",
-    "nyc": "^15.1.0",
-    "tape": "^4.11.0",
-    "tape-spawn": "^1.4.2",
-    "ts-node": "^10.4.0",
-    "typescript": "^4.5.4"
+    "@types/node": ">=16.11.7",
+    "@types/semver": ">=7.1.0",
+    "@types/tape": ">=4.2.27",
+    "@types/tmp": ">=0.2.0",
+    "@typescript-eslint/eslint-plugin": ">=5.0.0 <6.0.0",
+    "@typescript-eslint/parser": ">=5.0.0 <6.0.0",
+    "coveralls": ">=3.0.0",
+    "eslint": ">=7.12.1 <8.0.0",
+    "eslint-config-standard": ">=16.0.3 <17.0.0",
+    "eslint-plugin-import": ">=2.25.0",
+    "eslint-plugin-node": ">=11.1.0",
+    "eslint-plugin-promise": ">=5.0.0 <7.0.0",
+    "nyc": ">=15.0.0",
+    "tape": ">=4.11.0",
+    "tape-spawn": ">=1.0.0",
+    "ts-node": ">=10.0.0",
+    "typescript": ">=4.2.2 <5.0.0"
```

Comment:
Typescript, eslint and their plugins here are restricted only because they have issues on node.js 12 and 14. There are failures due to the use of `??` operator or its variants. On the other hand there are no issues on the latest node.js.

Not sure how to handle this. I guess node.js 12 and 14 are past EOL now so it may be ok to just remove the restriction. The issue is that our tests fail, so we would have to patch `package.json` in our tests.

Another possibility might be to just drop official v12 and v14 support.